src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -240,6 +273,8 @@ public class ColumnFamilyStore implements ColumnFamilyStoreMBean
     private final String oldMBeanName;
     private volatile boolean valid = true;
 
+    private Memtable.Factory memtableFactory;
```

Comment:
Should `memtableFactory` be `volatile`? It does look like we call `reload()` from `Schema.alterTalbe()` and a few other places that could be problematic...